hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/HoodieStreamerUtils.java

```diff
@@ -116,16 +128,32 @@ static Option<JavaRDD<HoodieRecord>> createHoodieRecords(HoodieStreamer.Config c
 
           return new CloseableMappingIterator<>(ClosableIterator.wrap(itr), rec -> {
             InternalRow row = (InternalRow) deserializer.deserialize(rec).get();
-            String recordKey = builtinKeyGenerator.getRecordKey(row, baseStructType).toString();
-            String partitionPath = builtinKeyGenerator.getPartitionPath(row, baseStructType).toString();
-            return new HoodieSparkRecord(new HoodieKey(recordKey, partitionPath),
-                HoodieInternalRowUtils.getCachedUnsafeProjection(baseStructType, targetStructType).apply(row), targetStructType, false);
+            try {
+              String recordKey = builtinKeyGenerator.getRecordKey(row, baseStructType).toString();
+              String partitionPath = builtinKeyGenerator.getPartitionPath(row, baseStructType).toString();
+              return Either.left(new HoodieSparkRecord(new HoodieKey(recordKey, partitionPath),
+                  HoodieInternalRowUtils.getCachedUnsafeProjection(baseStructType, targetStructType).apply(row), targetStructType, false));
+            } catch (Exception e) {
+              if (!shouldErrorTable) {
+                throw e;
+              }
+              try {
+                return Either.right(HoodieAvroUtils.avroToJsonString(rec, false));
+              } catch (IOException ex) {
+                throw new RuntimeException(ex);
+              }
+            }
           });
+
         });
       } else {
         throw new UnsupportedOperationException(recordType.name());
       }
-      return records;
+      if (shouldErrorTable) {
+        errorTableWriter.get().addErrorEvents(records.filter(Either::isRight).map(Either::asRight).map(evStr -> new ErrorEvent<>(evStr,
```

Comment:
should we also check if `errorTableWriter` is present or not?

Reply:
```
boolean shouldErrorTable = errorTableWriter.isPresent();
```

Reply:
decided to add a config as well so 
```
boolean shouldErrorTable = errorTableWriter.isPresent() && props.getBoolean(ERROR_ENABLE_VALIDATE_RECORD_CREATION.key(), ERROR_ENABLE_VALIDATE_RECORD_CREATION.defaultValue());
```